source/common/grpc/grpc_web_filter.cc

```diff
@@ -130,5 +168,47 @@ Http::FilterTrailersStatus GrpcWebFilter::encodeTrailers(Http::HeaderMap& traile
   return Http::FilterTrailersStatus::Continue;
 }
 
+void GrpcWebFilter::setupStatTracking(const Http::HeaderMap& headers) {
+  Router::RouteConstSharedPtr route = decoder_callbacks_->route();
+  if (!route || !route->routeEntry()) {
+    return;
+  }
+
+  const Router::RouteEntry* route_entry = route->routeEntry();
+  Upstream::ThreadLocalCluster* cluster = cm_.get(route_entry->clusterName());
+  if (!cluster) {
+    return;
+  }
+  cluster_ = cluster->info();
+
+  if (headers.Path() != nullptr && headers.Path()->value().c_str() != nullptr) {
+    std::vector<std::string> parts = StringUtil::split(headers.Path()->value().c_str(), '/');
+    if (parts.size() != 2) {
+      return;
+    }
+    grpc_service_ = parts[0];
+    grpc_method_ = parts[1];
+    do_stat_tracking_ = true;
+  }
+}
+
+void GrpcWebFilter::chargeStat(const Http::HeaderMap& headers) {
```

Comment:
code basically copied from bridge filter and https://github.com/lyft/envoy/blob/master/source/common/grpc/common.cc#L27. Can we share?